src/libopensc/asn1.c

```diff
@@ -89,9 +89,10 @@ int sc_asn1_read_tag(const u8 ** buf, size_t buflen, unsigned int *cla_out,
 			 * unsigned int */
 			return SC_ERROR_INVALID_ASN1_OBJECT;
 
+	} else {
+		if (left == 0)
+			return SC_ERROR_INVALID_ASN1_OBJECT;
 	}
-	if (left == 0)
-		return SC_ERROR_INVALID_ASN1_OBJECT;
 	/* parse length byte(s) */
 	len = *p & 0x7f;
 	if (*p++ & 0x80) {
```

Comment:
On 2/10/2015 3:21 AM, Frank Morgner wrote:
> In src/libopensc/asn1.c <https://github.com/OpenSC/OpenSC/pull/375#discussion_r24398396>:
>
>>  	}
>> -	if (left == 0)
>> -		return SC_ERROR_INVALID_ASN1_OBJECT;
>>  	/* parse length byte(s) */
>>  	len = *p & 0x7f;
>>  	if (*p++ & 0x80) {
>
> Both versions are semantically identical, see coverity scan CID 13628. It's hard to argue about the beauty of the code. I, personally, think that superfluous operations should be omitted.
>

My personal option, some minor performance can be sacrificed for code readability, especially in code that
gets enhanced and modified by a committee of volunteers (current and future) none of them who have the hardware
to test all the code.

> —
> Reply to this email directly or view it on GitHub <https://github.com/OpenSC/OpenSC/pull/375/files#r24398396>.
>

-- 

  Douglas E. Engert  <DEEngert@gmail.com>